Products/ZenModel/DataRoot.py

```diff
@@ -876,6 +876,19 @@ def _updateEmailNotifications(self, REQUEST):
     def zmanage_editProperties(self, REQUEST=None, redirect=False):
         """Handle our authentication mechanism
         """
+        curuser = self.dmd.ZenUsers.getUser().getId()
+        curpasswd = REQUEST.get('curPasswd')
+
+        if not self.dmd.ZenUsers.authenticateCredentials(curuser, curpasswd):
+            messaging.IMessageSender(self).sendToBrowser(
+                'Error',
+                'Confirmation password is empty or invalid. Please'+
+                ' confirm your password for security reasons.',
+                priority=messaging.WARNING
+            )
+            return self.callZenScreen(REQUEST)
+
+
```

Comment:
I think all of this should be performed only for requests from the browser, otherwise, `zendmd` scripts be blocked because it doesn't provide a _REQUEST_ object.